pkg/alertmanager/operator.go

```diff
@@ -98,18 +98,23 @@ func New(c prometheusoperator.Config, logger log.Logger) (*Operator, error) {
 		crdclient: crdclient,
 		logger:    logger,
 		queue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "alertmanager"),
-		config:    Config{Host: c.Host, ConfigReloaderImage: c.ConfigReloaderImage, AlertmanagerDefaultBaseImage: c.AlertmanagerDefaultBaseImage},
+		config: Config{
+			Host:                         c.Host,
+			ConfigReloaderImage:          c.ConfigReloaderImage,
+			AlertmanagerDefaultBaseImage: c.AlertmanagerDefaultBaseImage,
+			Namespace:                    c.Namespace,
+		},
 	}
 
 	o.alrtInf = cache.NewSharedIndexInformer(
 		&cache.ListWatch{
-			ListFunc:  o.mclient.MonitoringV1().Alertmanagers(api.NamespaceAll).List,
-			WatchFunc: o.mclient.MonitoringV1().Alertmanagers(api.NamespaceAll).Watch,
+			ListFunc:  o.mclient.MonitoringV1().Alertmanagers(o.config.Namespace).List,
+			WatchFunc: o.mclient.MonitoringV1().Alertmanagers(o.config.Namespace).Watch,
```

Comment:
How do you still do `api.NamespaceAll` ?

Reply:
I made sure to use the `api.NamespaceAll` variable as the default value for the flag.